pkg/build/metrics/prometheus/metrics.go

```diff
@@ -1,43 +1,59 @@
 package prometheus
 
 import (
-	kselector "k8s.io/apimachinery/pkg/labels"
 	"strings"
 
 	"github.com/golang/glog"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kselector "k8s.io/apimachinery/pkg/labels"
+
 	buildapi "github.com/openshift/origin/pkg/build/apis/build"
 	internalversion "github.com/openshift/origin/pkg/build/generated/listers/build/internalversion"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
 const (
-	separator               = "_"
-	buildSubsystem          = "openshift_build"
-	terminalBuildCount      = "terminal_phase_total"
-	terminalBuildCountQuery = buildSubsystem + separator + terminalBuildCount
-	activeBuildCount        = "running_phase_start_time_seconds"
-	activeBuildCountQuery   = buildSubsystem + separator + activeBuildCount
+	separator                 = "_"
+	buildSubsystem            = "openshift_build"
+	terminalBuildCount        = "terminal_phase_total"
+	terminalBuildCountQuery   = buildSubsystem + separator + terminalBuildCount
+	failedBuildCount          = "failed_phase_total"
+	failedBuildCountQuery     = buildSubsystem + separator + failedBuildCount
+	activeBuildCount          = "running_phase_start_time_seconds"
+	activeBuildCountQuery     = buildSubsystem + separator + activeBuildCount
+	newPendingBuildCount      = "new_pending_phase_creation_time_seconds"
+	newPendingBuildCountQuery = buildSubsystem + separator + newPendingBuildCount
+	errorBuildReason          = "BuildPodError"
 )
 
 var (
-	// decided not to have a separate counter for failed builds, which have reasons,
-	// vs. the other "finished" builds phases, where the reason is not set
 	terminalBuildCountDesc = prometheus.NewDesc(
-		buildSubsystem+separator+terminalBuildCount,
-		"Counts total teriminal builds by phase",
+		terminalBuildCountQuery,
+		"Counts total successful/aborted builds by phase",
```

Comment:
Why not count all builds by phase?

Reply:
That is one of the key parts of this change.  The failed/error phases have been broken off and lumped into one phase, so that we don't need a "phase" label for those, and can use 1 label for "reason".

Conversely, for complete/cancelled, disitnguish those two by phase, but they don't need a "reason";

Ultimately, space savings.

Reply:
I think that it will be clearer for the end user + provide a cleaner api + be easier for us to support/maintain if we don't do fancy logic/special casing here but provide straight-through statistics.

Reply:
Yeah, this specialization has certainly arisen, from my perspective, given previously guidance/threads around needing to be very optimal/minimal with the data introduced.

That so far has been the priority.

So point noted, at least somewhat (not sure of the degree to which is is easier/cleaner/etc.), but given my clarification, I at least still say we leave this as is.

Reply:
@jim-minter some of this complexity is also driven by the need to collect per-build metrics for new/active/running builds (so we can have information about when they started) compared with completed/failed/error builds which we just roll up as a single aggregate data point (or in the case of failed builds, a few labeled data points).

It does feel like for a future iteration we could extend the pattern @gabemontero established for the reasons, resulting in a single metric:  buildCount with two labels: phase and reason (reason would be empty for any build not in a "failed" phase).

We'd do all the aggregation in the collector, so we'd only be sending up a handful of data points(one data point per phase-type, plus one per failure reason), so having two labels per data point shouldn't be egregious.

Does that align with what you were thinking @jim-minter ?

Reply:
@bparees yes.

Reply:
@gabemontero thoughts?  i'm willing to entertain it as bug fix post-devcut :)

Reply:
just to be clear, prior pre-merged versions of the first pass at these metrics had just that, and was ultimately rejected.

but as noted previously, continual iteration / refinement is the mode of operation here, so a revisit followed by acceptance instead of reject the next time we entertain it is conceivable.

so "yes" ... but *absolutely* a post-devcut, or post-3.7, item.